app/batid/services/imports/import_bdnb7.py

```diff
@@ -52,14 +63,16 @@ def import_bdnb7_bdgs(dpt):
     cols = candidates[0].keys()
 
     with open(buffer.path, "r") as f:
-        with connection.cursor() as cursor:
+        with transaction.atomic():
```

Comment:
Par defaut, les connection.cursor() sont en autocommit et ne créent pas de transaction. Pour cela, il faut utiliser transaction.atomic() et en cas d'erreur, tout ce qui est dans le try est rollback.

voir la [doc](https://docs.djangoproject.com/en/4.2/topics/db/transactions/#managing-database-transactions) : "Django’s default behavior is to run in autocommit mode. Each query is immediately committed to the database, unless a transaction is active."

et cette [partie](https://docs.djangoproject.com/en/4.2/topics/db/transactions/#controlling-transactions-explicitly) sur les transactions.

Reply:
je m'en suis rendu compte car lors de mes tests, j'ai fait une requête qui a crashé, mais la requête sensée être dans la même transaction n'a pas été rollbackée.